### ISSUE TITLE
GML-independent modelling of basic types

### ISSUE THREAD

**Issue by TatjanaKutzner** [opengeospatial/CityGML-3.0CM#16]
In CityGML 2.0, the data type _TransformationMatrix4x4_ is defined as restriction of the GML 3.2.1 data type _doubleList_. In UML, _TransformationMatrix4x4_ can be modelled as subclass of the class _doubleList_ from the GML 3.2.1 UML model. This leads to the following problem: _doubleList_ is a data type which is specific to the GML language. The CityGML UML model, however, is language-independent; GML is only one of several possible encodings. Thus, it is actually not correct, to use a language-specific data type in our language-independent CityGML UML model.
![grafik](https://user-images.githubusercontent.com/5949401/45300888-d5266400-b50f-11e8-8378-a68b2b3d9cae.png)

**Comment by TatjanaKutzner** [comment]
To solve this I suggest to define a language-independent _DoubleList_ type from which _TransformationMatrix4x4_ is derived. During the encoding to GML, the encoding of this _DoubleList_ class can be suppressed. Furthermore, a tagged value is added to _TransformationMatrix4x4_ which defines that the data type is to be encoded as restriction of the GML base type _gml:doubleList_.
![grafik](https://user-images.githubusercontent.com/5949401/45301064-609ff500-b510-11e8-8c65-f935543cf8a8.png)
The difference to above, where the GML _doubleList_ type is used as super class, is only minimal, also the resulting encoding will be identical. However, now the type is not taken from the GML 3.2.1 UML model, but is defined language-independently.

**Comment by 3DXScape** [comment]
This also seems to be necessary to break the dependency on GML concepts. This idea of supplying a private copy of the GML concept with a "note" to the developers of encodings that for the GML encoding, the native GML construct can be used. This is no different than adopting native representations of strings, numbers, truth values, and other primitives in individual encodings.

**Comment by TatjanaKutzner** [comment]
We discussed in the web conference on 17 May 2019 that the suggested definition of the type _DoubleList_ is fine. 
However, we will check how INSPIRE approaches this issue.

**Comment by TatjanaKutzner** [comment]
I found another dependency from GML. The type `MeasureOrNilReasonList` is a GML-specific type. The type is used in the class `AbstractBuilding`:
![grafik](https://user-images.githubusercontent.com/5949401/76339969-1cef9c80-62fb-11ea-838a-7b8b21d6d813.png)

The dependency can be solved in a similar way as above by defining a language-independent `MeasureOrNilReasonList` type. 
However, the type brings several more classes along which also need to be defined in the UML model:
![grafik](https://user-images.githubusercontent.com/5949401/76403846-d3985f00-6385-11ea-8be4-37e823a87c27.png)
I set tagged values for each class which suppresses them from being encoded. Only `MeasureOrNilReasonList` will be mapped to the corresponding GML type in the encoding.
